packages/vscode-extension/src/debugging/RadonCDPProxyDelegate.ts

```diff
@@ -246,19 +285,34 @@ export class RadonCDPProxyDelegate implements CDPProxyDelegate {
     throw new Error("Source map URL schemas other than `data` and `http` are not supported");
   }
 
-  private async handleScriptParsed(command: IProtocolCommand): Promise<IProtocolCommand> {
+  private async handleScriptParsed(
+    command: IProtocolCommand,
+    tunnel: ProxyTunnel
+  ): Promise<IProtocolCommand> {
     const { sourceMapURL, url, scriptId } = command.params as Cdp.Debugger.ScriptParsedEvent;
     if (!sourceMapURL) {
       return command;
     }
 
+    this.isBreakpointSettingPaused = false;
+    this.dispatchPendingBreakpointCommands();
+
     try {
       const sourceMapData = await this.getSourceMapData(sourceMapURL);
       const isMainBundle = sourceMapData.sources.some((source: string) =>
         source.includes("__prelude__")
       );
 
       this.sourceMapRegistry.registerSourceMap(sourceMapData, url, scriptId, isMainBundle);
+
+      if (isMainBundle) {
+        this.HMRSocket.send(
```

Comment:
does it mean we will only be notified of the changes made in the main bundle then? For example, when we reload a module that has already been reloaded we won't be getting that?

Why is the replace statement there? Give example and explain

Reply:
No, this isn't confusing that HMR server requires registration. What I was asking there was that we only register for the main bundle and not for other endpoints. I'd expect that we should register for all scripts that gets eventually parsed, as for example the HRM may load additional files that didn't exist in the original main bundle in which case we'd never react to the changes made in those files.

Reply:
As for my "alternative approach" what I was proposing was to make it so that we skip when debugger pauses on a breakpoint that isn't set in the most recent version of the code. For this purpose, we wouldn't even need to maintain the HMR connection as we could tell this just purely based on the source maps.

Have you checked if this issue you're solving here happens when connecting chome debugger?